lib/archethic/transaction_chain/transaction/validation_stamp/ledger_operations.ex

```diff
@@ -280,26 +294,61 @@ defmodule Archethic.TransactionChain.Transaction.ValidationStamp.LedgerOperation
       %{uco: uco_balance, token: tokens_received} = ledger_balances(inputs)
       %{uco: uco_to_spend, token: tokens_to_spend} = total_to_spend(ops)
 
-      new_unspent_outputs = [
-        %UnspentOutput{
-          from: change_address,
-          amount: uco_balance - uco_to_spend,
-          type: :UCO,
-          timestamp: timestamp
-        }
-        | new_token_unspent_outputs(
-            tokens_received,
-            tokens_to_spend,
-            change_address,
-            inputs,
-            timestamp
-          )
-      ]
+      %{uco: consumed_uco_utxos, token: consumed_token_utxos} =
+        get_inputs_to_consume(inputs, uco_to_spend, tokens_to_spend)
```

Comment:
I think the logic of consumed inputs is that if we take the consumed inputs and apply the movement of the transaction, it will result the same as the unspent outputs.
For now as we still put all the utxos in the unspent outputs, I think we should have all the inputs as well in the consumed inputs.
And then on the part 2 of the AEIP21 we will use your calculation of which inputs are really used.

Also in part 2 this algorithm could be done by the genesis pool before sending the inputs to the validation nodes

Reply:
But the part 1 is also to make sure the list of consumed inputs is the right one, before to actually using it.

Reply:
Hum for me not really.
I think the goal of the first part is to ensure that the genesis pool is storing well what we expect it to store. It should delete the consumed inputs and create the new unspent outputs.
If you don't put all the inputs in the consumed inputs, it means that you could have unspent outputs in the stamp that are already known and stored by the genesis pool. So the genesis pool will have a check to create the utxo only if it does not already exists.
But when we'll go on part 2 this check will not be needed anymore so we will update how the genesis pool store the unspent outputs which will make the part 1 less usefull

Reply:
The genesis pool need also the list of consumed inputs, to return the only right list of real unspent outputs later on the phase 2.
the genesis would still need to use the unspent outputs in the stamp to reallocate the real new unspent outputs.

Reply:
I think we should prepare the algorithm and storage to experiment it before real usage.